packages/jest-changed-files/src/hg.js

```diff
@@ -17,59 +18,56 @@ const env = Object.assign({}, process.env, {
   HGPLAIN: 1,
 });
 
-type Options = {|
-  lastCommit?: boolean,
-  withAncestor?: boolean,
-|};
-
-function findChangedFiles(cwd: string, options: Options): Promise<Array<Path>> {
-  return new Promise((resolve, reject) => {
-    let args = ['status', '-amn'];
-    if (options && options.withAncestor) {
-      args.push('--rev', 'ancestor(.^)');
-    } else if (options && options.lastCommit === true) {
-      args = ['tip', '--template', '{files%"{file}\n"}'];
-    }
-    const child = childProcess.spawn('hg', args, {cwd, env});
-    let stdout = '';
-    let stderr = '';
-    child.stdout.on('data', data => (stdout += data));
-    child.stderr.on('data', data => (stderr += data));
-    child.on('error', e => reject(e));
-    child.on('close', code => {
-      if (code === 0) {
-        stdout = stdout.trim();
-        if (stdout === '') {
-          resolve([]);
-        } else {
-          resolve(
-            stdout
-              .split('\n')
-              .map(changedPath => path.resolve(cwd, changedPath)),
-          );
-        }
-      } else {
-        reject(code + ': ' + stderr);
+const adapter: SCMAdapter = {
+  findChangedFiles: async (
+    cwd: string,
+    options: Options,
+  ): Promise<Array<Path>> => {
+    return new Promise((resolve, reject) => {
+      let args = ['status', '-amnu'];
+      if (options && options.withAncestor) {
+        args.push('--rev', 'ancestor(.^)');
```

Comment:
@cpojer any idea what `ancestor(.^)` does? it seems to be used only inside www. HG documentation is very complicated and honestly after 15 min of reading it i was still not sure :)
if this is returning current cahnges + the latest commit, `--rev .^` should be enough